### DISCUSSION TITLE
Using a antler variable as counter

### DISCUSSION THREAD

**Discussion by Bahbv** [statamic/cms#6164] [Q&A] [answered]
Hi guys, 

Im trying to use a variable to keep the count of the amount of collection posts my replicator block called "Overzicht" spits out.  
I want to do this so a user could split up an overview of a collection(ie: blog posts) with other blocks without worrying about which one it reads out.  

The code i've got;

            {{ offset = 0 }}

            {{ collection:blogposts as="posts" }}

                {{ page_builder }}

                    {{ if type !== 'overzicht' }}
                        {{ partial src="page_builder/{type}" }}
                    {{ /if }}

                    {{ if type == 'overzicht'}}
                        {{ posts offset="{{ offset }}" limit="3" }}

                            {{ offset+= 1}}

                            <div class="c12-xs">{{ offset }} | {{ url }}</div>
                        {{ /posts }}
                    {{ /if }}

                {{ /page_builder }}

            {{ /collection:blogposts }}

It seems like it doesn't work like this, the offset variable gets reset every 'pagebuilder' block.  
Is it even possible what i am trying to do? Is there a other way?

**Comment by JohnathonKoster** [comment, marked as answer]
What you have looks valid to me and should work. What version of Statamic are you using? I could replicate this behavior on 3.3.12 and below (but not on 3.3.13).

**Comment by Bahbv** [comment]
Hi Johnathon thanks for replying,  

Weird you've got it working on 3.3.12, thats the version I am using. It could be that you didn't quite understand what I'm trying to do, so I added a screenshot. The 4, 5, 6 in red is what I was expecting. But like I said, the offset variable keeps resetting to 0 after each pagebuilder/replicator block.  

![Schermafbeelding 2022-06-07 145116](https://user-images.githubusercontent.com/12693117/172385232-e59ee38c-09a2-4575-abcc-c6f4a4304c32.jpg)

**Reply by JohnathonKoster** [reply]
> Weird you've got it working on 3.3.12,

I may have worded my original message badly. I was not able to get it working on versions 3.3.12 and below, but was able to make this work on 3.3.13 (which includes a number of variable scope fixes).

Can you try on 3.3.13 to see if it resolves your issue?

**Reply by Bahbv** [reply]
Oh wow, small miscommunication. You could replicate the behaviour I had, got it. It works perfect on 3.3.13 indeed. Thanks for maintaining this great CMS!
